app/views.py

```diff
@@ -89,9 +119,13 @@ def faq():
     """Renders the faq page."""
     return render_template('faq.html', title=' | FAQ')
 
-@app.route('/progressreport')
-def progressreport():
+@app.route('/progressreport/<int:knewbie_id>')
+def progressreport(code):
```

Comment:
```suggestion
def progressreport(knewbie_id):
```
Should have the same name as in the routing.